src/backend/distributed/utils/maintenanced.c

```diff
@@ -266,7 +267,16 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		if (DistributedDeadlockDetectionTimeoutFactor != -1.0)
 		{
 			StartTransactionCommand();
-			foundDeadlock = CheckForDistributedDeadlocks();
+
+			/*
+			 * We don't want to run the deadlock checks if there exists
+			 * any version mistmatch.
+			 */
+			if (CheckCitusVersion(reportLevel))
```

Comment:
DEBUG1 seems fine. We really don't care about running deadlock detection when the version is behind, since there's no way the user can start a query (hopefully).